src/keycache.ts

```diff
@@ -0,0 +1,46 @@
+import {Mutex} from 'async-mutex';
+import {PeerId} from "./@types/libp2p";
+import {bytes, bytes32} from "./@types/basic";
+
+/**
+ * Storage for static keys of previously connected peers.
+ */
+class Keycache {
+  private mutex: Mutex;
```

Comment:
why do you need mutex, storage should be thread safe by default

Reply:
Handshakes are async and I guess a lot of connections could go through at the same time. Those were my thoughts... Not sure what do you mean by default?

Reply:
It doesn't matter if they are async, as long as you are dealing with javascript object it will be thread safe (methods on storage will be executed sequentially). It makes sense using mutex if you are dealing with native database api or maybe using worker threads

Reply:
https://stackoverflow.com/questions/7238586/do-i-need-to-be-concerned-with-race-conditions-with-asynchronous-javascript/7238663#7238663

Reply:
The answer is pretty vague, especially the second part, but it could be summed up as "only if using some native code". I guess I've been reading the wrong things.